lib/neo4j-server/cypher_session.rb

```diff
@@ -23,6 +23,10 @@ def initialize(data_url, connection)
       # @return [Faraday]
       # @see https://github.com/lostisland/faraday
       def self.create_connection(params, url = nil)
+        faraday_options = params.delete(:faraday_options) || params.delete('faraday_options') || {}
+        adapter = (faraday_options[:adapter] || faraday_options['adapter'] || :net_http_persistent).to_sym
```

Comment:
Rather than just extracting the `adaptor` option I was actually thinking of the `faraday_options` just being passed as a whole to `Faraday.new` so that we don't have to support individual options but can let users change whatever faraday options that they want.  I think it would make the code a bit cleaner too.  Does that seem reasonable?